### DISCUSSION TITLE
GetComponent<I인터페이스> 할때, 해당 인터페이스 컴포넌트를 불러오지못함.

### DISCUSSION THREAD

**Discussion by JeongDongBaek** [naverz/zepeto-studio-kor#236] [5. Q&A]
제페토 기본 요소들을 참고하여 닿으면 UI제공 및 델리게이트 실행을 구현하고 있습니다.
![image](https://user-images.githubusercontent.com/33086479/150483653-d5176667-8846-4028-bceb-0e182e7d019c.png)

인터페이스는 위와 같습니다.

![image](https://user-images.githubusercontent.com/33086479/150483724-f15b852f-d944-4a0c-9422-15529bd3ee66.png)
인터페이스를 구현하는 스크립트는 위와 같습니다.
위 스크립트에서 Start (이벤트함수) 는 잘 먹히고 있습니다. 인터페이스 누락 경고도 잘 뜨고있구요...

문제는 해당 클래스(인터페이스를 구현하고 있는)를  타 클래스에서  불러올때 문제가 발생합니다.
![image](https://user-images.githubusercontent.com/33086479/150484211-4bfaef33-9834-44c5-8a93-d70378c909e5.png)

해당 인터페이스를 구현하고 있는 오브젝트의 경우, 해당 인터페이스에 해당되는 정보들을 받아오고 싶은데, 인터페이스의 존재를 null로 파악하고 있습니다.

typeof를 써서 더 명확하게 지시내리고 싶으나 타입스크립트에서는 기본 자료형말고는 typeof를 쓸 수 없다고 경고가 나오는 것같습니다.

방법이 있을까요?

![image](https://user-images.githubusercontent.com/33086479/150484394-23965bf9-a63b-4b10-9376-1f52e0db2018.png)

**Comment by shlifedev** [comment]
안녕하세요. 질문해주신 케이스에 대해서는 아쉽게도 아직 지원되지 않습니다.

제페토 스크립트는 C#의 API와 통신하여 실제로는 C#의 API를 호출하는 구조입니다.
`typescript` 에서 `interface` 를 선언 후 상속해도 C#에서 `interface`가 선언되어 있는것은 아니기 때문에
의도하신대로 동작하지 않는 것 같습니다. 
    
해당 부분은 추후 개선 사항에 추가하도록 하겠습니다. 
감사합니다.
